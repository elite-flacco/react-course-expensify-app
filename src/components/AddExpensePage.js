import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { addExpense } from '../actions/expenses';
import { startAddExpense } from '../actions/expenses';

export class AddExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.startAddExpense(expense);
        this.props.history.push('/');
    };
    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h2 className="page-header__title">Add expense</h2>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm
                        onSubmit={this.onSubmit}
                    />
                </div>
            </div>
        );
    }
}
// const AddExpensePage = (props) => (
//     <div>
//         <h1>Add expense</h1>
//         <ExpenseForm 
//             onSubmit={(expense) => {
//                 // props.dispatch(addExpense(expense));
//                 props.onSubmit(expense);
//                 props.history.push('/');
//             }}
//         />
//     </div>
// )

const mapDispatchToProps = (dispatch) => ({
    startAddExpense: (expense) => dispatch(startAddExpense(expense))
})

// export default connect()(AddExpensePage);
export default connect(undefined, mapDispatchToProps)(AddExpensePage);