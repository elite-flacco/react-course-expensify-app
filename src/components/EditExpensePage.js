import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { editExpense, removeExpense, startRemoveExpenses } from '../actions/expenses';

export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.editExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    }
    onClick = () => {
        this.props.startRemoveExpenses( {id: this.props.expense.id} );
        this.props.history.push('/');
    }
    render() {
        return (
            <div>
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}    
                />
                <button onClick={this.onClick}>Remove</button>
            </div>   
        )
    };
}
// const EditExpensePage = (props) => {
//     console.log(props);
//     return (
//         <div>
//             Editing the expense with the id of {props.expense.id}
//             <ExpenseForm
//                 expense={props.expense}
//                 onSubmit={(expense) => {
//                     props.dispatch(editExpense(props.expense.id, expense));
//                     props.history.push('/');
//                 }}    
//             />
//             <button onClick={() => {
//                 props.dispatch(removeExpense( {id: props.expense.id} ));
//                 props.history.push('/');
//             }}>Remove</button>
//         </div>
//     )
// }

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    }

}

const mapDispatchToProps = (dispatch) => ({
    editExpense: (id, expense) => dispatch(editExpense(id, expense)),
    startRemoveExpenses: (id) => dispatch(startRemoveExpenses(id))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);