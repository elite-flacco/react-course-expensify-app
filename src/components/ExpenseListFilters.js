import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';
import { DateRangePicker } from 'react-dates';

export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    };

    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    };
    onFocusChange = (calendarFocused) => {
        this.setState(() => ({ calendarFocused }));
    };
    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value);
    }
    onSortChange = (e) => {
        if (e.target.value === 'date') {
            this.props.sortByDate();
        } else {
            this.props.sortByAmount();
        }

    }
    render() {
        return (
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input
                            className="text-input"
                            type="text"
                            placeholder="Search expenses"
                            value={this.props.filter.text}
                            onChange={this.onTextChange}
                        />
                    </div>
                    <div className="input-group__item">
                        <select
                            className="select"
                            type="text"
                            value={this.props.filter.sortBy}
                            onChange={this.onSortChange}>
                            <option value="date">Date</option>
                            <option value="amount">Amount</option>
                        </select>
                    </div>
                    <div className="input-group__item">
                        <DateRangePicker
                            startDate={this.props.filter.startDate}
                            endDate={this.props.filter.endDate}
                            onDatesChange={this.onDatesChange}
                            focusedInput={this.state.calendarFocused}
                            onFocusChange={this.onFocusChange}
                            isOutsideRange={(day) => false}
                            showClearDates={true}
                        />
                    </div>
                </div>
            </div>
        );
    }

}
// const ExpenseListFilters = (props) => (
//     <div>
//         <input 
//             type="text" 
//             value={props.filter.text} 
//             onChange={(e) => {
//                 props.dispatch(setTextFilter(e.target.value));
//             }}
//         />
//         <select 
//             type="text"
//             value={props.filter.sortBy}
//             onChange={(e) => {
//                 if (e.target.value === 'date') {
//                     props.dispatch(sortByDate());
//                 } else {
//                     props.dispatch(sortByAmount());
//                 }

//             }}>
//             <option value="date">Date</option>
//             <option value="amount">Amount</option>
//         </select>
//     </div>
// );

const mapStateToProps = (state) => {
    return {
        filter: state.filter
    };
};

const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount()),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate)),
})

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);