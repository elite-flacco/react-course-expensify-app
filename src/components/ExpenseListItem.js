import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = (props) => (
    <div>
        <Link to={`/edit/${props.id}`}>
            <h1>{props.description}</h1>
        </Link>
        <p>{props.amount}</p>
        <p>Create at: {props.createdAt}</p>
    </div>
);

export default ExpenseListItem;

// <h1><a href={`/edit/${props.id}`}>{props.description}</a></h1> => this causes a full page refresh