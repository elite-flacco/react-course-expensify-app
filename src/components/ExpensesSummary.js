import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expenseCount, expenseTotal}) => {
    return (
        <div>
            <p>There are {expenseCount} expenses</p>
            <p>Sum of the expenses is {numeral(expenseTotal).format('$0,0.00')}</p>
        </div>

    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filter);

    return {
        expenseCount: visibleExpenses.length,
        expenseTotal: selectExpensesTotal(visibleExpenses)
    }
    
}

export default connect(mapStateToProps)(ExpensesSummary);