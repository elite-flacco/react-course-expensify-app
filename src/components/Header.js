import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
    <header>
        <h1>Expensify App</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Go back to homepage</NavLink>
        <NavLink to="/create" activeClassName="is-active">Create expense</NavLink>
        <NavLink to="/help" activeClassName="is-active">Need help</NavLink>
    </header>
)

export default Header;