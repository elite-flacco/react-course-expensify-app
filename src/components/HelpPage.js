import React from 'react';

const HelpPage = () => (
    <div>
        need help!!!
    </div>
)
export default HelpPage;