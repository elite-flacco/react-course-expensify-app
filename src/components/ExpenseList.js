import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>No Expenses</p>
            ) : (
                props.expenses.map((expense) => (
                <ExpenseListItem key={expense.id} {...expense}/>
                // <ExpenseListItem key={expense.id} description={expense.description}/>
            )))
        }
    </div>
)

// const ConnectedExpenseList = connect((state) => {
//     return {
//         expenses: state.expenses
//     };
// })(ExpenseList);

// export default ConnectedExpenseList;

// const mapStateToProps = (state) => {
//     return {
//         expenses: state.expenses,
//         filter: state.filter
//     };
// };

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filter),
    };
};


export default connect(mapStateToProps)(ExpenseList);