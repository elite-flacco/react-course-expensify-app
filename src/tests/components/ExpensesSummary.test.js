import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={2} expenseTotal={2345} />);
    expect(wrapper).toMatchSnapshot();
});
