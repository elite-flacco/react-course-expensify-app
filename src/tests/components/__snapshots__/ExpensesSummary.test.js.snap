// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary 1`] = `
<div>
  <p>
    There are 
    2
     expenses
  </p>
  <p>
    Sum of the expenses is 
    $2,345.00
  </p>
</div>
`;
